best-practices.md

```diff
@@ -26,23 +28,18 @@ All rules below should be followed by contributors to this repo. Contributors sh
   - Note that future versions of WDL will likely cause a change to this convention.
     - We plan to deprecate the `ncpu` param in favor of accessing the runtime section directly (`n_cores=~{task.runtime.cpu}`)
 - Tasks which assume a file and any accessory files (e.g. a BAM and a BAI) have specific extensions and/or are in the same directory should *always* follow the conventions laid out in the `localize_files_task` example (see `template/task-examples.wdl.template`)
-  - This is to accomodate as many backends as possible
+  - This is to accommodate as many backends as possible
 - output file names should *always* be determined with either the `outfile_name` parameter or the `prefix` parameter.
   - `outfile_name` should be preferred if no downstream tasks/tools rely on the file name/extension
   - tasks with multiple outputs should always use the `prefix` convention
-- After the input sorting rules in `style-guide.md` have been applied, follow the below rules for further sorting.
+- After the input sorting rules in `sprocket lint` have been applied, follow the below rules for further sorting.
   - "sample" files come before "reference" files
   - If present, `use_all_cores` should be the last `Boolean` in its block
   - the `ncpu` parameter comes before inputs that allocate memory, which come before inputs that allocate disk space
     - This block of 2-3 inputs should come after all other inputs.
 - Most tasks should have a default `maxRetries` of 1
-  - Certain tasks are prone to intermittent failure (often if an internet connection is involved) and can have a higher default `maxRetries`. This value should not exceed 3.
-- There are lower bounds for resource allocation
-  - Memory should be allocated a minimum of 4gb
-  - Disk size should be allocated a minimum of 10gb
-  - If the task is multi-cored, it should use at least 2 cpu by default
-  - These bounds were selected somewhat arbitrarily, but consistency is important for quickly identifying our light-weight tasks
-  - These bounds are subject to change pending a more empirical investigation
+  - Certain tasks are prone to intermittent failure (often if an internet connection is involved) and can have a higher default `maxRetries`.
+- If a task uses multiple cores or is multithreaded, then at least 2 cpu should be specified.
 - All tasks should have an output
```

Comment:
per convo in slack, this can be removed. Then I think we'll be good to merge this